app/bundles/LeadBundle/Controller/LeadController.php

```diff
@@ -1394,10 +1394,48 @@ public function emailAction($objectId = 0)
                             empty($email['fromname']) ? null : $email['fromname']
                         );
 
+                        // Attachments
+                        $assets = [];
+                        $assetsIds  = (!empty($email['assetAttachments'])) ? $email['assetAttachments'] : [];
+                        foreach($assetsIds as $assetId) {
+                            $model = $this->factory->getModel('asset');
+                            $asset = $model->getEntity($assetId);
+                            if ($asset->isPublished()) {
+                                $asset->setUploadDir($this->factory->getParameter('upload_dir'));
+                                $assets[$asset->getId()] = $asset;
+                            }
+                        }
+
+                        if (!empty($assets)) {
+                            foreach ($assets as $asset) {
+                                if(is_file($asset->getFilePath())) {
+                                    $mailer->message->attach(\Swift_Attachment::fromPath(
+                                        $asset->getFilePath(),
+                                        $asset->getMime()
+                                    )->setFilename($asset->getOriginalFileName()));
+                                }
+                            }
+                        }
+
+
+                        if(isset($email['attachments']) && !empty($email['attachments'])) {
+                            foreach($email['attachments'] as $attachment) {
+                                $identifier = time();
+                                $attachment->move($this->get('kernel')->getRootDir() . "/../media/attachments", $identifier . "-" . $attachment->getClientOriginalName());
+                                $mailer->message->attach(\Swift_Attachment::fromPath(
+                                    $this->get('kernel')->getRootDir() . "/../media/attachments/" . $identifier . "-" . $attachment->getClientOriginalName()
+                                )->setFilename($attachment->getClientOriginalName()));
+                            }
+                        }
+
                         // Set Content
                         $mailer->setBody($email['body']);
                         $mailer->parsePlainText($email['body']);
 
+                        // Set BCC/CC
```

Comment:
I think these lines need to check values, If I let it empty, then return error for me 

>  500 Internal Server Error - PHP Warning - count(): Parameter must be an array or an object that implements Countable 

```
/app/bundles/EmailBundle/Helper/MailHelper.php:1156 at
Mautic\CoreBundle\ErrorHandler\ErrorHandler -> handleError ( '2', 'PHP Warning - count(): Parameter must be an array or an object that implements Countable', '/var/www/html/app/bundles/EmailBundle/Helper/MailHelper.php', '1156', array('addresses' => null, 'name' => null) )
/app/bundles/EmailBundle/Helper/MailHelper.php:1156 at count ( null )
/app/bundles/LeadBundle/Controller/LeadController.php:1436 at Mautic\EmailBundle\Helper\MailHelper -> setBcc ( null )
```